### ISSUE TITLE
Inconsistent isAnotherUserEditing results

### ISSUE THREAD

**Issue by payten** [steveswinsburg/gradebookNG#155]
I've noticed that the isAnotherUserEditing endpoint isn't behaving consistently.  When a change is detected, we get the result:

GET: http://localhost:8080/direct/gbng/isotheruserediting/test-site.json?_=1437702110654

```
 {"entityPrefix": "gbng", "gbng_collection": [
 {
   "assignmentId": 25,
   "lastUpdated": 1437702144445,
   "lastUpdatedBy": "Sakai Administrator",
   "studentUuid": "7cc60e0e-3167-490d-9261-a8cfe20cae3a",
   "entityReference": "\/gbng",
   "entityURL": "http:\/\/localhost:8080\/direct\/gbng"
 }
 ]}
```

6 seconds later we poll again however the changes are no longer visible:
GET: http://localhost:8080/direct/gbng/isotheruserediting/test-site.json?_=1437702110655

```
 {"entityPrefix": "gbng", "gbng_collection": []}
```

I've also seen one user notified of changes, while the other use doesn't receive any notifications.  :confused:

**Comment by payten** [comment]
Hm... the cache is definitely shared between the users.. after logging the hashCode of the `cache`:

```
 ****** cache for: payten hashCode: 857700099
 ****** cache for: admin hashCode: 857700099
 ****** cache for: payten hashCode: 857700099
 ****** cache for: admin hashCode: 857700099
 ...
```

**Comment by payten** [comment]
Aha! I think this is our culprit:

```
 if(notifications != null) {
     notifications.remove(currentUserId);
```

It deletes the user's notifications out of the cache sometimes before the other user gets a chance to see it.  @steveswinsburg I might need you to double check my logic.

**Comment by payten** [comment]
Fixed in PR #156.
